### ISSUE TITLE
[CodeGen React UI] Support parametrization of nginx port

### ISSUE THREAD

**Issue by Feelas** [opea-project/GenAIExamples#503]
In the nginx.conf for CodeGen React UI, there is a hardcoded port 80.
https://github.com/opea-project/GenAIExamples/blob/09a3196324b6b9094852c9a990d2a9346f779cda/CodeGen/docker/ui/react/nginx.conf#L2

Supporting parametrization of nginx port is not a purely cosmetic matter, since it causes issues on unpriviliged deployments on K8s/OpenShift.

It would be good to both:
- switch the default from port 80 to a known-working port outside of 1-1023 range, since it is is privileged by default and requires specific securityContext (which would be a terrible thing to use).
- support parametrizing nginx port in an easy way, which doesn't require rebuilding the image (see the docs for upstream image used: [https://hub.docker.com/_/nginx](https://hub.docker.com/_/nginx) -> **Using environment variables in nginx configuration (new in 1.19)**)

The fastest way to fix the issue would be to just switch port 80 to 8080.

**Comment by lianhao** [comment]
Agreed. When fixing this, please also fix https://github.com/opea-project/GenAIComps/blob/main/comps/third_parties/nginx/src/nginx.conf.template at the same time.

**Comment by xiguiw** [comment]
> * witch the default from port 80 to a known-working port outside of 1-1023 range, since it is is privileged by default and requires specific securityContext (which would be a terrible thing to use).

Per @lianhao. It is not permitted to run with privileged/root for some public K8S

**Comment by xiguiw** [comment]
@jaswanth8888 

It decreases the security of OPEA. This is a critical issue.
Pleased resolve is ASAP.

**Comment by yinghu5** [comment]
Link [Support parametrization of nginx port #1456](https://github.com/opea-project/GenAIComps/pull/1456)
Just Remind @jaswanth8888 @ZePan110 , hopefully the PR can fix the issues.

**Comment by xiguiw** [comment]
React UI set as non-root is traced here: #517

**Comment by xiguiw** [comment]
@sgurunat 

[remind] Please help on this issue.  Thanks!
